jwst/ramp_fitting/ramp_fit_step.py

```diff
@@ -242,11 +246,43 @@ def process(self, input):
                 self.save_opt, readnoise_2d, gain_2d, self.algorithm,
                 self.weighting, max_cores, dqflags.pixel, suppress_one_group=self.suppress_one_group)
 
-        # Save the OLS optional fit product, if it exists
-        if opt_info is not None:
-            opt_model = create_optional_results_model(input_model, opt_info)
-            self.save_model(opt_model, 'fitopt', output_file=self.opt_name)
+            # To calculate the readnoise variance for weighting, ramp_fitting
+            # will be run a second time - this time with the DO_NOT_USE flags
+            # removed from the UNDERSAMP-flagged pixels. To reduce execution time,
+            # fitting will effectively be performed only on the UNDERSAMP-flagged
+            # pixels, by flagging all other pixels as DO_NOT_USE.
+
+            # First create an intermediate model ('_W' for weighting) to
+            # input to a 2nd ramp_fitting call.
+            input_model_W = copy.copy(input_model)
```

Comment:
The input model gets modified during `ramp_fit`, such as NaNs getting set in the SCI array, as well as DO_NOT_USE and SATURATION flags possibly getting set in the group DQ array.  Should this copy occur before the first call to `ramp_fit`?

Reply:
Good point. Made that change.